l1-contracts-foundry/script/DeployL1.s.sol

```diff
@@ -0,0 +1,703 @@
+            console.log("Multicall3 deployed at:", contractAddress);
+        }
+    }
+
+    function deployVerifier() internal {
+        address contractAddress = deployViaCreate2(type(Verifier).creationCode);
+        console.log("Verifier deployed at:", contractAddress);
+        addresses.stateTransition.verifier = contractAddress;
+    }
+
+    function deployDefaultUpgrade() internal {
+        address contractAddress = deployViaCreate2(type(DefaultUpgrade).creationCode);
+        console.log("DefaultUpgrade deployed at:", contractAddress);
+        addresses.stateTransition.defaultUpgrade = contractAddress;
+    }
+
+    function deployGenesisUpgrade() internal {
+        address contractAddress = deployViaCreate2(type(GenesisUpgrade).creationCode);
+        console.log("GenesisUpgrade deployed at:", contractAddress);
+        addresses.stateTransition.genesisUpgrade = contractAddress;
+    }
+
+    function deployValidatorTimelock() internal {
+        uint32 executionDelay = uint32(config.contracts.validatorTimelockExecutionDelay);
+        bytes memory bytecode = abi.encodePacked(
+            type(ValidatorTimelock).creationCode,
+            abi.encode(config.deployerAddress, executionDelay, config.eraChainId)
+        );
+        address contractAddress = deployViaCreate2(bytecode);
+        console.log("ValidatorTimelock deployed at:", contractAddress);
+        addresses.validatorTimelock = contractAddress;
+    }
+
+    function deployGovernance() internal {
+        bytes memory bytecode = abi.encodePacked(
+            type(Governance).creationCode,
+            abi.encode(
+                config.contracts.governorAddress,
+                config.contracts.governanceSecurityCouncilAddress,
+                config.contracts.governanceMinDelay
+            )
+        );
+        address contractAddress = deployViaCreate2(bytecode);
+        console.log("Governance deployed at:", contractAddress);
+        addresses.governance = contractAddress;
+    }
+
+    function deployTransparentProxyAdmin() internal {
+        ProxyAdmin proxyAdmin = new ProxyAdmin();
+        console.log("Transparent Proxy Admin deployed at:", address(proxyAdmin));
+        proxyAdmin.transferOwnership(addresses.governance);
+        addresses.transparentProxyAdmin = address(proxyAdmin);
+    }
+
+    function deployBridgehubContract() internal {
+        address bridgehubImplementation = deployViaCreate2(type(Bridgehub).creationCode);
+        console.log("Bridgehub Implementation deployed at:", bridgehubImplementation);
+        addresses.bridgehub.bridgehubImplementation = bridgehubImplementation;
+
+        bytes memory bytecode = abi.encodePacked(
+            type(TransparentUpgradeableProxy).creationCode,
+            abi.encode(
+                bridgehubImplementation,
+                addresses.transparentProxyAdmin,
+                abi.encodeCall(Bridgehub.initialize, (config.deployerAddress))
```

Comment:
I would try to distinguish between `owner` and `deployer` here. The `owner` can often be set to the already deployed governance, i.e. the fact that `owner = deployer` holds only for local developemtn